Library/Source/NativeEngine.cpp

```diff
@@ -1081,6 +1198,71 @@ namespace babylon
         delete textureData;
     }
 
+    Napi::Value NativeEngine::Impl::CreateFrameBuffer(const Napi::CallbackInfo& info)
+    {
+        const auto textureData = info[0].As<Napi::External<TextureData>>().Data();
+        uint16_t width = static_cast<uint16_t>(info[1].As<Napi::Number>().Uint32Value());
+        uint16_t height = static_cast<uint16_t>(info[2].As<Napi::Number>().Uint32Value());
+        bgfx::TextureFormat::Enum format = static_cast<bgfx::TextureFormat::Enum>(info[3].As<Napi::Number>().Uint32Value());
+        int samplingMode = info[4].As<Napi::Number>().Uint32Value();
+        bool generateStencilBuffer = info[5].As<Napi::Boolean>();
+        bool generateDepth = info[6].As<Napi::Boolean>();
+        bool generateMipMaps = info[7].As<Napi::Boolean>();
+
+        bgfx::FrameBufferHandle frameBufferHandle{};
+        if (generateStencilBuffer && !generateDepth)
+        {
+            throw std::exception{ /* Does this case even make any sense? */ };
+        }
+        else if (!generateStencilBuffer && !generateDepth)
+        {
+            frameBufferHandle = bgfx::createFrameBuffer(width, height, TEXTURE_FORMAT[format], BGFX_TEXTURE_RT);
+        }
+        else
+        {
+            auto depthStencilFormat = bgfx::TextureFormat::D32;
+            if (generateStencilBuffer)
+            {
+                depthStencilFormat = bgfx::TextureFormat::D24S8;
+            }
+
+            std::array<bgfx::TextureHandle, 3> textures
+            {
+                bgfx::createTexture2D(width, height, generateMipMaps, 1, TEXTURE_FORMAT[format], BGFX_TEXTURE_RT),
```

Comment:
Why  2 2D textures here?
Also, you have to check if the format is supported for rendering into
https://bkaradzic.github.io/bgfx/bgfx.html?highlight=istexturevalid#_CPPv4N4bgfx14isTextureValidE8uint16_tb8uint16_tN13TextureFormat4EnumE8uint64_t

Reply:
Good point. I used two color textures here because I was following (somewhat blindly) the pattern from the BGFX bloom and deferred lighting examples (https://github.com/bkaradzic/bgfx/blob/master/examples/21-deferred/deferred.cpp#L470-L489). It seems to work here with two, so I've switched it to that, will go up with the next push. But do we know why the examples use 3?

Reply:
It's a gbuffer. The pixel shader writes to both target (like albedo in 1RT and normal in the other one)

Reply:
Makes sense. I've added asserts to check whether the format of the textures is supported, as I don't think we currently have a back-up plan for what to do if a format isn't supported.

Reply:
We should create an issue for that. I had some issues on Base3D Android because of that and had to use some formats as replacement.

Reply:
Added a line to #6. Hopefully BGFX's cross-platform capabilities and constraints will shield us from some issues like this; but if it doesn't, the asserts and the checkbox should ensure we don't forget.